File: day_1/part_2/solution.rb
frequencies = []

File.open("../input.txt", "r").each_line do |line|
  frequencies << line.chomp.to_i
end

total = 0
series = {}

while true do
  frequencies.each do |reading|
    if series[total]
      puts total # found duplicate
      exit
    else
      series[total] = true
    end

    total += reading
  end
end

puts "None found"

File: day_1/part_2/solution.py
import sys

solution = 0
series = {}

frequencies = []
with open("../input.txt", "r") as file:
    for line in file:
        frequencies.append(int(line))

while True:
    for reading in frequencies:
        if solution in series:
            print(solution) # found duplicate
            sys.exit()
        else:
            series[solution] = True

        solution += reading
